phastSim/phastSim.py

```diff
@@ -641,23 +809,196 @@ def sampleMutation(self, allele, rates, rand):
             "You should not have got here - there was a bug somewhere or some unlucky sampling in the machine error area")
         exit()
 
+    def sampleDeletion(self, rate, rand):
 
+        # TODO use numpy sampling from geometric distribution if available
+        # this rand needs to be fresh and not at all related to the random number
+        # coming from sampling a position on the genome.
+        
+        if rate <= 0 or rand <= 0:
+            return 1
+
+        return 1 + int(np.log(rand)/np.log(rate))
+
+    def deleteNodes(self, rand, node, remaining_deletions, level):
```

Comment:
I understand what this function does - and it seems correct to me.
Could you however add a comment explaining the idea behind this function?
The way this works is not trivial, so a nice explanation could really be useful in future!

Reply:
Great! Sure, I will add something like:
# This function tidies up the genomeTree after a deletion has been sampled, by traversing the tree in preorder, 
# and ignoring all nodes that appear to the left of the first node or once the total number of nodes have been deleted. 
# Nodes are 'deleted' by setting their allele to "-" and setting all of their rates to 0, creating this new empty node at the correct level if necessary.